### ISSUE TITLE
Do not display #nsfw posts at all to users under 18yo

### ISSUE THREAD

**Issue by Flaburgan** [diaspora/diaspora#7315]
We do not respect the french law for years and I feel uncomfortable about it. We should not display any #nsfw posts if the date of birth of the user is set and he is under 18yo (as the majority can differ depending of the country maybe this can be a setting). I know that technically this can be easily avoided but it would legally put the responsibility on the user and not the podmin.

**Comment by Globulle** [comment]
I agree with @Flaburgan .
Besides I think the feature *Do not display #nsfw posts at all* should also be made available to adult users. I know some people who fight against their porn addiction and seeing posts with explicit content in their stream doesn't help at all, even if the picture is not immediately visible. You can ignore people who annoy you, why coudn't you also remove disturbing posts from your stream?

**Comment by goobertron** [comment]
Well, but..... #nsfw does *not* mean '18+'. It means 'something which might get you into trouble if your strict boss walks past your desk when you're looking at Diaspora'. It has nothing to do with adult content or extreme content. I know most people use it to post porn - sorry, 'art', but that's not what the tag means, so I don't think it should be automatically blocked for people under 18. You'd need another standard for that. And, conversely, I don't think you can be said to be breaking the law by allowing younger people to view things that aren't 'safe for work'.

Perhaps a new standard, and tag, is needed for 'extreme content'.

**Comment by Globulle** [comment]
@goobertron : You are right, in theory. But even in the definition found on [wikipedia](https://en.wikipedia.org/wiki/Not_safe_for_work) this tag is clearly associated to *nudity, intense sexuality, profanity or disturbing content*. And I assume that *disturbing content* is a priori not appropriate to young public, even it is not directly sexually-related.
And speaking about the law, I personally think that if something seems like a right thing to do, there is absolutely no problem to be ahead of the law. This is what we do concerning privacy. :-)

**Comment by denschub** [comment]
Honestly? Not sure. We don't respect the law anyway since we don't force people to provide their birthday, and that's for a reason. We've had the same decision when we limited the birthyear field to 13 years+ for "respecting" COPPA, but still, since it's an optional field, that protection is basically worthless.

Same goes for adding a stream filter (which, ohmygosh, another condition in the stream query), which would increase the complexity by far and I highly doubt people will run into this (on Gerasporas DB, there are exactly three users who are younger than 18 years - all of them are company profiles). Maybe the solution is to reword the NSFW shield to "possibly inappropriate content", which would work for any European laws I was able to find (including the French laws).
